src/main/java/uk/gov/pay/directdebit/mandate/services/gocardless/GoCardlessMandateStateUpdater.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.pay.directdebit.mandate.services.gocardless;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GoCardlessOrganisationId;
+import uk.gov.pay.directdebit.mandate.dao.MandateDao;
+import uk.gov.pay.directdebit.mandate.model.GoCardlessMandateId;
+
+import javax.inject.Inject;
+
+import static java.lang.String.format;
+import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.GOCARDLESS;
+
+public class GoCardlessMandateStateUpdater {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(GoCardlessMandateStateUpdater.class);
+
+    private final MandateDao mandateDao;
+    private final GoCardlessMandateStateCalculator goCardlessMandateStateCalculator;
+
+    @Inject
+    public GoCardlessMandateStateUpdater(MandateDao mandateDao, GoCardlessMandateStateCalculator goCardlessMandateStateCalculator) {
+        this.mandateDao = mandateDao;
+        this.goCardlessMandateStateCalculator = goCardlessMandateStateCalculator;
+    }
+
+    public void updateState(GoCardlessMandateId goCardlessMandateId, GoCardlessOrganisationId goCardlessOrganisationId) {
+        goCardlessMandateStateCalculator.calculate(goCardlessMandateId, goCardlessOrganisationId)
+                .ifPresent(state -> {
```

Comment:
Do we need to log anything if its empty?

Reply:
Not sure. Do we need to know if we received an event that didn’t require us to update the state? I do want to give the logging here more thought though.

Reply:
I think that information could be useful, especially if we ever encounter a scenario where we should be updating the state but aren't for a particular event, at least there might be a log line telling us that such an event doesn't require an update (which I assume would be less frequent outcome than those that do?).

Reply:
OK, I’ll add something